src/processor/tx-monitor-queue-consumer.service.ts

```diff
@@ -0,0 +1,149 @@
+import { Injectable, Logger, OnApplicationBootstrap, OnModuleDestroy } from '@nestjs/common';
+import { InjectQueue, Processor, WorkerHost } from '@nestjs/bullmq';
+import { DelayedError, Job, Queue } from 'bullmq';
+import { GraphUpdateJobState, IGraphUpdateJob } from '#app/interfaces/graph-update-job.interface';
+import { BlockchainService } from '#app/blockchain/blockchain.service';
+import { MILLISECONDS_PER_SECOND } from 'time-constants';
+import { SchedulerRegistry } from '@nestjs/schedule';
+import { BlockchainConstants } from '#app/blockchain/blockchain-constants';
+import { ReconnectionCacheMgrService } from '#app/cache/reconnection-cache-mgr.service';
+import { ReconnectionServiceConstants } from '#app/constants';
+import { ITxStatus } from '#app/interfaces/tx-status.interface';
+import { Graph } from '@dsnp/graph-sdk';
+
+const CAPACITY_EPOCH_TIMEOUT_NAME = 'capacity_check';
+
+@Injectable()
+@Processor(ReconnectionServiceConstants.TX_MONITOR_QUEUE_NAME)
+export class TxMonitorQueueConsumerService extends WorkerHost implements OnApplicationBootstrap, OnModuleDestroy {
+  private logger: Logger;
+
+  public async onApplicationBootstrap() {
+    await Promise.all([this.blockchainService.api.isReady, this.blockchainService.apiPromise.isReady]);
+    await this.blockchainService.checkCapacity();
+    this.txMonitorQueue.resume();
+  }
+
+  public onModuleDestroy() {
+    try {
+      this.schedulerRegistry.deleteTimeout(CAPACITY_EPOCH_TIMEOUT_NAME);
+    } catch (e) {
+      // ignore error
+    }
+  }
+
+  constructor(
+    @InjectQueue(ReconnectionServiceConstants.TX_MONITOR_QUEUE_NAME) private readonly txMonitorQueue: Queue,
+    @InjectQueue(ReconnectionServiceConstants.GRAPH_UPDATE_QUEUE_NAME) private readonly graphUpdateQueue: Queue,
+    private readonly cacheManager: ReconnectionCacheMgrService,
+    private readonly blockchainService: BlockchainService,
+    private readonly schedulerRegistry: SchedulerRegistry,
+  ) {
+    super();
+    this.logger = new Logger(this.constructor.name);
+  }
+
+  async process(job: Job<any, any, string>): Promise<void> {
+    this.logger.debug(`Processing monitorUpdateTxns job ${job.id}`);
+    const jobTxns = await this.cacheManager.getAllTxnsForJob(job.data.id);
+    const sourceJob: Job<IGraphUpdateJob, any, string> | undefined = await this.graphUpdateQueue.getJob(job.data.id);
```

Comment:
Will this job exists in `graphUpdateQueue` if `removeOnSuccess` is true for that queue? since tx is generated and job is successful it might be removed ?